Dockerfile

```diff
@@ -1,93 +1,131 @@
 ###########################################################
-# Base container with all necessary deps
-# Buster slim python 3.7 base image.
-FROM python:3.7-slim-buster as base
-ENV HTTP_PORT 8080
-
-RUN groupadd -r geoadmin \
-    && useradd -r -s /bin/false -g geoadmin geoadmin \
-    && mkdir -p /logs && chown geoadmin:geoadmin /logs \
-    # install relevent packages
-    && apt-get update \
-    && apt-get install -y binutils libproj-dev gdal-bin \
-    # the following line contains debug tools that can be removed later
+# Container that contains basic configurations used by all other containers
+# It should only contain variables that don't change or change very infrequently
+# so that the cache is not needlessly invalidated
+FROM python:3.9-slim-bullseye as base
+ENV HTTP_PORT=8080
+ENV USER=geoadmin
+ENV GROUP=geoadmin
+ENV INSTALL_DIR=/opt/service-stac
+
+RUN groupadd -r ${GROUP} \
+    && useradd -r -s /bin/false -g ${GROUP} ${USER} \
+    && mkdir -p ${INSTALL_DIR}/logs && chown ${USER}:${GROUP} ${INSTALL_DIR}/logs
+
+WORKDIR ${INSTALL_DIR}
```

Comment:
I would not change the workdir here in the base image, it is then quite confusing when reading the file. Workdir should change only in the debug and production image at the end of debug and production image. See also comment about the pipfiles. The goal of the builder is to have only what is needed to run the application on the production images while the builder image has everything needed to build/compile/transpile/install the application.

Reply:
Because it is impossible to use paths relative to the `${INSTALL_DIR}` if I don't specify the WORKDIR

Reply:
Yes I think this would make sense and be a bit clearer where stuff are copied.

Reply:
Ok perfect then I will change that.